org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnMessageSendName.java

```diff
@@ -27,14 +28,19 @@
 public class CompletionOnMessageSendName extends MessageSend implements CompletionNode {
 
 	public boolean nextIsCast;
+	public boolean cursorIsToTheLeftOfTheLParen;
 
-	public CompletionOnMessageSendName(char[] selector, int start, int end, boolean nextIsCast) {
+	public CompletionOnMessageSendName(char[] selector, int start, int end) {
+		this(selector, start, end, setup -> {/* use defaults */});
+	}
+
+	public CompletionOnMessageSendName(char[] selector, int start, int end, Consumer<CompletionOnMessageSendName> setup) {
```

Comment:
instead of using a Consumer for setup it would be easier to understand and maintain if you directly pass the initial values to the constructor. probably "cursorIsToTheLeftOfTheLParen" misses a "t" at the end and could be final.